### PR TITLE
Add Interface for Autocomplete Response

### PR DESCRIPTION
**Pull request by geoffcoutts**

### CODE REVIEW

File: src/core/sayt.ts [hunk 1]
```diff
@@ -98,6 +98,42 @@ export interface ProductSearchConfig {
   productSort?: any;
 }
 
+export interface AutocompleteResponse {
+  status: {
+    code: number;
+    internalCode: number;
+    message: 'string';
```
Comment (juliamlim) [inline src/core/sayt.ts]: Take out quotes
```suggestion
    message: string;
```

File: src/core/sayt.ts [hunk 2]
```diff
@@ -98,6 +98,42 @@ export interface ProductSearchConfig {
...
+    stats: {
+      navigationCount: number;
+      searchCount: number;
+      productCount: number;
+      autocompleteResponse: number;
+      productSearchResponse: number;
+    }
+    searchTerms: AutocompleteSearchTerm[] | null;
+    navigations: AutocompleteNavigation[] | null;
+    products: any;
```
Comment (juliamlim) [inline src/core/sayt.ts]: This is probably going to always be an array so let's do:
```suggestion
    products: any[];
```
or if we want to account for a more standardized model in the future we can create a Product(data?) interface that is just typed `any` for now

File: src/core/sayt.ts [hunk 3]
```diff
@@ -98,6 +98,42 @@ export interface ProductSearchConfig {
   productSort?: any;
 }
 
+export interface AutocompleteResponse {
+  status: {
+    code: number;
+    internalCode: number;
+    message: 'string';
+    additionalInfo: any;
```
Comment (juliamlim) [inline src/core/sayt.ts]: We should probably keep the same structure as the additionalInfo below
```suggestion
    additionalInfo: {
      [key: string]: any;
    };
```

Comment (geoffcoutts) [reply src/core/sayt.ts]: The one below was explicitly set to that in the API files, however, this one wasn't and can be `null` instead.

Comment (juliamlim) [reply src/core/sayt.ts]: Ah cool 👍

File: src/core/sayt.ts (hunk 2 above)
Comment (geoffcoutts) [reply src/core/sayt.ts]: Have added an interface for products and set it to 
```
products: AutocompleteProducts[] | null
```

**Pull request merged**